src/app/qgsmapcanvasdockwidget.cpp

```diff
@@ -336,6 +353,12 @@ void QgsMapCanvasDockWidget::syncViewCenter( QgsMapCanvas *sourceCanvas )
   mBlockExtentSync = false;
 }
 
+void QgsMapCanvasDockWidget::syncSelection( QgsMapCanvas *sourceCanvas )
+{
+  QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( sourceCanvas->currentLayer() );
```

Comment:
`sourceCanvas` could be checked before using it. Just to be sure it's not null.

Reply:
Actually I think one could directly access `mMainCanvas` in here and drop the parameter.